### ISSUE TITLE
Rename "Song Editor" to "Sequencer"

### ISSUE THREAD

**Issue by unfa** [LMMS/lmms#121]
The titles says it all.
Let's have shorter and more universal names,

**Comment by badosu** [comment]
Is this still a thing? If so, could someone point out how this change would affect LMMS holistically? I guess this would not just be a change to the window label but a fundamental terminology change.

**Comment by tresf** [comment]
> Is this still a thing?

Yeah, it's still under (dormant) discussion.  Most of the devs agreed for better names and this topic still remains relevant. :)

**Comment by badosu** [comment]
Oh, just noticed that this discussion is more elaborated at #120. I'll check there.

**Comment by tresf** [comment]
Yeah, the renames should probably be organized and consolidated since nomanclature in general tends to affect a lot of the code.

**Comment by tresf** [comment]
I'd like to resurrect this thread.  Most were in favor of changing it back in January 2014 when it was originally proposed.  This is a large refactoring task.

**Comment by Wallacoloo** [comment]
@tresf Can this be as simple as a translation file edit? (I believe we can have an english translation file)

**Comment by tresf** [comment]
> Can this be as simple as a translation file edit

Well, it *could*... We could make the class called [`_tmp`](https://github.com/LMMS/lmms/issues/2149#issuecomment-118210155) too I suppose... :smiley:

**Comment by Wallacoloo** [comment]
@tresf Point taken - translation files might not be the best way to tackle this - indeed, I can think of multiple reasons why they aren't currently suited for the job.

We should come at it from an angle where it's easy to rename it again in the future though, right? I mean, it's pretty much impossible to make all of the *sources* trivially renamable, but from the UI perspective, it should amount to a single string constant being changed - right?

**Comment by tresf** [comment]
> but from the UI perspective, it should amount to a single string constant being changed

I think this is one of those all-or-nothing scenarios where we just need to pick a new permanent name (or alternatively decide to keep what we have).

I don't think we'll ever have easy class renaming. :)  From a UI perspective, you're right, the front end could be changed without changing the underlying class names, but I just think it would be hard for developers to discuss and would make the codebase confusing.

**Comment by Wallacoloo** [comment]
@tresf hmm. I haven't dug into the code too far yet, but if we were to reduce this to one string constant per editor, I would suspect it allows for more code reuse.

For example, we could add a virtual method, `editorName()` to the `Editor` class & implement that in all the subclasses. Then the Editor class can be responsible for setting the Window title instead of implementing that functionality in *every* subclass, we can create the `show/hide xxx` options in the view menu programatically - `tr ("Show/Hide {1}").arg(editor->name())`... I'm sure there are more areas that would benefit which I cannot yet think of.

**Comment by tresf** [comment]
As part of a pruning effort, this `enhancement` request is archived into a dedicated "Better Workflow" checklist here #4877.
